PWGCF/Flow/Tasks/flowZdcTask.cxx

```diff
@@ -676,12 +662,88 @@ struct FlowZdcTask {
     float commonSumZpa = (zdcread.energyCommonZPA());
     float sumZN = (sumZNC) + (sumZNA);
     float sumZP = (sumZPC) + (sumZPA);
-    histos.fill(HIST("zPos"), collision.posZ());
-    histos.fill(HIST("T0Ccent"), collision.centFT0C());
-    histos.fill(HIST("ZNAcoll"), znA);
-    histos.fill(HIST("ZNCcoll"), znC);
-    histos.fill(HIST("ZPAcoll"), zpA);
-    histos.fill(HIST("ZPCcoll"), zpC);
+
+    int itsTracks = 0, glbTracks = 0;
+    for (const auto& track : tracks) {
+      // Track Selection
+      if (track.hasITS()) {
+        itsTracks++;
+      }
+      if (!track.isGlobalTrack()) {
+        continue;
+      }
+      if ((track.pt() < minPt) || (track.pt() > maxPt)) {
+        continue;
+      }
+      histos.fill(HIST("ZposVsEta"), collision.posZ(), track.eta());
+      histos.fill(HIST("EtaVsPhi"), track.eta(), track.phi());
+      histos.fill(HIST("dcaXYvspT"), track.dcaXY(), track.pt());
+      glbTracks++;
+    }
+    bool skipEvent{false};
+    if (useMidRapNchSel) {
+      auto hMeanNch = ccdb->getForTimeStamp<TH1F>(paTHmeanNch.value, foundBC.timestamp());
+      auto hSigmaNch = ccdb->getForTimeStamp<TH1F>(paTHsigmaNch.value, foundBC.timestamp());
+      if (!hMeanNch) {
+        LOGF(info, "hMeanNch NOT LOADED!");
+        return;
+      }
+      if (!hSigmaNch) {
+        LOGF(info, "hSigmaNch NOT LOADED!");
+        return;
+      }
+      const int binT0M{hMeanNch->FindBin(normT0M)};
+      const double meanNch{hMeanNch->GetBinContent(binT0M)};
+      const double sigmaNch{hSigmaNch->GetBinContent(binT0M)};
+      const double nSigmaSelection{nSigmaNchCut * sigmaNch};
+      const double diffMeanNch{meanNch - glbTracks};
+      if (!(std::abs(diffMeanNch) < nSigmaSelection)) {
+        histos.fill(HIST("ExcludedEvtVsFT0M"), normT0M);
+        histos.fill(HIST("ExcludedEvtVsNch"), glbTracks);
+      } else {
+        skipEvent = true;
+      }
+    }
+    // Skip event based on number of Nch sigmas
+    if (!skipEvent) {
+      return;
+    }
+    std::vector<float> pTs;
+    std::vector<float> vecFD;
```

Comment:
Any reason for variable definition without any usage?
@vkucera Should not this be captured by the compiler? or not any longer?

Reply:
Yes, it should. No idea why it is not the case here.

Reply:
Seems like the compiler only complains about trivial types. See https://gcc.gnu.org/bugzilla/show_bug.cgi?id=55203